lxd/main_init.go

```diff
@@ -196,15 +202,15 @@ func (c *cmdInit) availableStorageDrivers(poolType string) []string {
 
 	// Check available backends.
 	for _, driver := range supportedDrivers {
-		if poolType == "remote" && !shared.StringInSlice(driver.Name, []string{"ceph", "cephfs"}) {
+		if poolType == poolTypeRemote && !shared.StringInSlice(driver.Name, []string{"ceph", "cephfs"}) {
 			continue
 		}
 
-		if poolType == "local" && shared.StringInSlice(driver.Name, []string{"ceph", "cephfs"}) {
+		if poolType == poolTypeLocal && shared.StringInSlice(driver.Name, []string{"ceph", "cephfs"}) {
 			continue
 		}
 
-		if poolType == "all" && driver.Name == "cephfs" {
+		if poolType == poolTypeAny && driver.Name == "cephfs" {
```

Comment:
@stgraber what lead me down this rabbit hole was this line. I was looking to reuse this `availableStorageDrivers` function for populating the list of driver type options available in the `lxd recover` tool when asking them to recover an unknown pool. However I was confused why `cephfs` is not returned for the `poolTypeAny` (formerly "all") option.

I hypothesise that this was because `cephfs` is not suitable for instance volumes and so followed the usage of this function back up the tree. 

However I was unable to find the "smoking gun" that would explain this behaviour. There seems to be some knowledge encoded in this logic, but I'm unsure why its there. Do you know why? I would like to make it clearer whats going on here.

Reply:
This is also relevant due to this bug fix https://github.com/lxc/lxd/pull/8893/commits/cf6d97a74f247e1b8047bc3cc5bbf0c63b0cb3b3

Reply:
Right, when showing a list of storage backends that the user can select (effectively for their default profile), we don't want to list `cephfs` as it can't hold instances.

Reply:
@stgraber yeah thats what I thought, but the bit that confused me was this bit:

https://github.com/lxc/lxd/blob/master/lxd/main_init_interactive.go#L527-L532

If the previous question was answered no, then it would be possible to use cephfs for the profile root disk storage pool right?

```
Do you want to configure a new local storage pool? (yes/no) [default=yes]: no
Do you want to configure a new remote storage pool? (yes/no) [default=no]: yes
Name of the storage backend to use (ceph, cephfs) [default=ceph]: cephfs
Create a new CEPHFS pool? (yes/no) [default=yes]: 
```